src/com/yandex/app/logic/HttpTaskServer.java

```diff
@@ -0,0 +1,321 @@
+
+                            subtask.setTaskId(id);
+                            taskManager.updateSubtask(subtask);
+                            sendText(exchange, "{\"Успех\": \"Подзадача обновлена\"}", 200);
+                        } catch (CollisionException e) {
+                            sendText(exchange, "{\"Ошибка\": \"Подзадача пересекается с другой\"}", 409);
+                        } catch (Exception e) {
+                            sendText(exchange, "{\"Ошибка\": \"Ошибка при обновлении подазадачи\"}", 500);
+                        }
+                    } else {
+                        sendText(exchange, "{\"Ошибка\": \"Неверный эндпоинт\"}", 404);
+                    }
+                    break;
+
+                case "DELETE":
+                    if (path.matches("/subtasks/\\d+")) {
+                        try {
+                            int id = Integer.parseInt(path.substring("/subtasks/".length()));
+                            taskManager.removeSubtaskById(id);
+                            sendText(exchange, "{\"Успех\": \"Подазадча уделена\"}", 200);
+                        } catch (NumberFormatException e) {
+                            sendText(exchange, "{\"Ошибка\": \"Неверный id задачи\"}", 400);
+                        } catch (Exception e) {
+                            sendText(exchange, "{\"Ошибка\": \"Ошибка при удалении подзадачи\"}", 500);
+                        }
+                    } else {
+                        sendText(exchange, "{\"Ошибка\": \"Неверный эндпоинт\"}", 404);
+                    }
+                    break;
+
+                default:
+                    sendText(exchange, "{\"Ошибка\": \"Неподдерживаемый метод\"}", 405);
+                    break;
+            }
+        }
+    }
+
+    private class UserHandler extends BaseHttpHandler {
+        @Override
+        public void handle(HttpExchange exchange) throws IOException {
+
+            String path = exchange.getRequestURI().getPath();
+
+            try {
+                if (path.startsWith("/history")) {
+                    String historyString = gson.toJson(taskManager.getHistory());
+                    sendText(exchange, historyString, 200);
+                } else if (path.startsWith("/prioritized")) {
+                    String prioritizedString = gson.toJson(taskManager.getPrioritizedTasks());
+                    sendText(exchange, prioritizedString, 200);
+                } else {
+                    sendText(exchange, "{\"Ошибка\": \"Неверный эндпоинт\"}", 404);
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+                sendText(exchange, "{\"Ошибка\": \"Внутренняя ошибка сервера\"}", 500);
+            }
+        }
+
+
+    }
+
+    public static void main(String[] args) {
+        TaskManager taskManager = new InMemoryTaskManager();
+
+        HttpTaskServer server = null;
```

Comment:
Все ссылочные типы по дефолту инициализируются значением null

```suggestion
        HttpTaskServer server;
```